api/ask_astro/chains/custom_llm_output_lines_parser.py

```diff
@@ -0,0 +1,38 @@
+from typing import Optional
```

Comment:
```suggestion
from __future__ import annotations
```

Reply:
Drive-by suggestion, this could be added automatically using and adding to pyproject.toml. This way it doesn't take up mental space when reviewing. (Probably worth updating target versions of the linters in the toml file to be 3.11 since the project requires 3.11+ too. That _might_ also help with the `Optional` typing syntax change via pyupgrade.)

```
[tool.ruff.isort]
required-imports = [ "from __future__ import annotations" ]
```